homework/src/index.js

```diff
@@ -31,17 +33,96 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
-    });
+
+
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { 'html': '<h3> HYF Repositories <h3>' });
+
+
+    const select = createAndAppend('select', header);
+    const div = createAndAppend('div', root, { 'id': 'div1' });
+    const div2 = createAndAppend('div', root, { 'id': 'div2' });
+
+
+    const promise = fetchJSON(url);
+    promise.then(data => {
+     
+      data.forEach((repo, index) => {
+        createAndAppend('option', select, { 'html': repo.name, 'value': index });
+        // console.log(repo.name, index);
+
+      });
+      select.addEventListener('change', (event) => {
+        const contrubutorsURL = data[event.target.value].contributors_url;
+        div2.innerHTML = '';
+        const secPromise = fetchJSON(contrubutorsURL);
+        secPromise.then(contdata => {
+          data.sort((a, b) => a.name.localeCompare(b.name, 'fr', { ignorePunctuation: true }));
+          contdata.forEach((contr) => {
+            container2(contr, div2);
+          });
+
+
+        });
+
+
+      });
+      select.addEventListener('change', (event) => {
+        div.innerHTML = '';
+
+        container1(data[event.target.value], div);
+
+
+      });
+
+
+    })
```

Comment:
There are a lot of unneeded (and as far as I'm concerned, unwanted) blank lines here.